test/cql-pytest/conftest.py

```diff
@@ -75,7 +76,13 @@ def cql(request):
         auth_provider=PlainTextAuthProvider(username='cassandra', password='cassandra'),
         ssl_context=ssl_context,
     )
-    return cluster.connect()
+    session = cluster.connect()
+    # Check if the tests run on Scylla version
+    test_config.IS_SCYLLA_VERSION = any(
```

Comment:
I think something is a bit off here. You rely on the "cql" fixture to set this variable (by the way, I don't understand why you put it in a separate file test_config and not just conftest.py), but who said the cql fixture is created first? :-(

In https://stackoverflow.com/questions/28179026/how-to-skip-a-pytest-using-an-external-fixture they suggest a different approach. Maybe you can add a function-scope autouse fixture (so it runs before every test) which uses a session-scope fixture for testing Scylla once. The function-scope autouse fixture will check if the test has the scylla_only marker, and if it does, skip this test.